index.bs

```diff
@@ -485,6 +516,158 @@ spec:csp3; type:grammar; text:base64-value
+  To determine <dfn>should request be blocked by integrity policy</dfn>, given a <a for=/>request</a> |request|,
+  do the following:
+
+  1. Let |policy container| be |request|'s <a>policy container</a>.
+  1. if |request|'s <a>integrity metadata</a> is not the empty string
+     and |request|'s <a for=request>mode</a> is either "cors" or "same-origin",
+     return "Allowed".
+  1. Let |policy| be |policy container|'s <a>integrity policy</a>.
+  1. Let |report policy| be |policy container|'s <a>report only integrity policy</a>.
+  1. If both |policy| and |report policy| are empty <a>integrity policy struct</a>s, return "Allowed".
+  1. Let |global| be |request|'s <a for=request>client</a>'s <a>global object</a>.
+  1. if |global| is not a <a>Window</a> nor a <a>WorkerGlobalScope</a>, return "Allowed".
+  1. Let |block| be a boolean, initially false.
+  1. Let |report block| be a boolean, initially false.
+  1. if |policy|'s <a>sources</a> <a for=list>contains</a> "`inline`"
+     and |policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |block| to true.
+  1. if |policy|'s <a>sources</a> <a for=list>contains</a> "`inline`"
+     and |report policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |report block| to true.
+  1. If |block| or |report block|, <a>report violation</a> with |request|, |block|, |report block|,
+     |policy| and |report policy|.
+  1. If |block|, return "Blocked". Otherwise, return "Allowed".
+ 
+  ### Report violations ### {#report-violations}
+
+  <pre class="idl">
+    [Exposed=Window]
+    interface IntegrityPolicyViolationReportBody : ReportBody {
+      [Default] object toJSON();
+      readonly attribute USVString documentURL;
+      readonly attribute USVString blockedURL;
+      readonly attribute USVString destination;
+      readonly attribute boolean   reportOnly;
+    };
+  </pre>
+
+  To <dfn>report violation</dfn> given a <a for=/>Request</a> |request|, a boolean |block|,
+  a boolean |report block|, an <a>integrity policy struct</a> |policy|,
+  and an <a>integrity policy struct</a> |report policy|, do the following:
+
+  1. Assert that |request|'s <a for=request>client</a> is not null.
+  1. Let |settings object| be |request|'s <a for=request>client</a>.
+  1. Let |global| be |settings object|'s <a>global object</a>.
+  1. Assert that |global| is a <a>Window</a> or a <a>WorkerGlobalScope</a>.
+  1. Let |url| be a <a for=/>URL</a>, initially null.
+  1. If |global| is a <a>Window</a>, set |url| to |global|'s <a>associated Document</a> <a>document URL</a>.
+  1. If |global| is a <a>WorkerGlobalScope</a>, set |url| to |global|'s <a for=WorkerGlobalScope>URL</a>.
+  1. Assert that |url| is a non-null <a for=/>URL</a>.
+  1. Let |document URL| be the result of <a>strip URL for use in reports</a> on |url|.
```

Comment:
```suggestion
  1. Let |documentURL| be the result of <a>strip URL for use in reports</a> on |url|.
```
Not using camelcase variables still applies quite a few times.

Reply:
done